docs/proposals/reduce-multitenancy-cost.md

```diff
@@ -0,0 +1,221 @@
+
+Neither the distributor nor the compactor needs to be modified as their current behavior is not impacted by the multiplexing tenancy.
+
+#### Read path
+
+Ingester and store-gateway use the tenant ID as a matcher when querying storage instead of opening the tenant's TSDB
+
+Ruler still work as-is (see <https://grafana.com/docs/mimir/latest/references/architecture/components/ruler/>): - In internal mode it runs its own distributor. Given the following proposal, we must ensure that we apply the same relabeling as in the rest of the write path - In remote mode, only the read path is changed. Indeed, filtering is performed by the ingester and store-gateway using the tenant ID.
+
+Alertmanager multitenancy still working as-is
+
+    - Ruler call the alertmanager with the tenant ID
+
+The tenants federation still works as-is.
+
+#### Limits
+
+Depending on the component in charge of enforcing the limits, we might have to adapt them:
+
+- distributor: no change expected
+- ingester: max_series_per_user and max_series_per_metrics have to be computed differently
+- querier/query-frontend/query-scheduler: no change expected
+- compactor: retention is per TSDB. It can be enforced in the read path by clamping the "start" of the query. Depending on the compute cost vs the storage cost, this might be more cost-efficient to keep the block for longer instead of processing them to discard the data that have exceeded their retention.
+- ruler: no change expected
+- alertmanager: no change expected
+
+#### Metrics
+
+Depending on the component in charge of metrics exposition, we might have to adapt them:
+
+- ingester: TSDB metrics aren't per user anymore.
+- compactor: metrics are per TSDB.
+- store-gateway: no change expected
+- distributor: no change expected
+- querier/query-frontend/query-scheduler: no change expected
+- ruler: no change expected
+- alertmanager: no change expected
+
+**Pros:**
+
+- The per-tenant overhead is greatly limited and can be considered negligible.
+
+**Cons:**
+
+- Unless modifications, all tenants within the same TSDB share some of the limits. This might be a blocker especially for the retention.
+- There is no easy migration back and forth between regular tenancy and tenancy multiplexing.
+- It is all or nothing as there is no way to define the mapping between the tenant ID and the TSDB.
+- Compactor blocks upload (<https://grafana.com/docs/mimir/latest/configure/configure-tsdb-block-upload/>) will not work properly as block are uploaded to LTS storage as-is. There is no way to add the metatenancy label without rewriting the block.
+- Compactor tenant deletion will not work as it relies on mapping the tenant to a dedicated TSDB.
+- Migrating a metatenant from one TSDB/cluster to another is not straightforward nor easy.
+
+### Proposal 2: Reduce the overhead by at least an order of magnitude
+
+Disclaimer: We list this proposal for the sake of completeness but our knowledge is limited and we don't know how realistic it is.
```

Comment:
It would be great to have a "tuning" page in the docs, with sections aimed at different scenarios.  
Some ideas for your scenario:
 * Reduce `-blocks-storage.tsdb.head-chunks-write-buffer-size-bytes`, default 4MB.  E.g. try 1MB, or 128KB.
 * Reduce `-blocks-storage.tsdb.stripe-size`, default 16384. E.g. try 256 or even 64.